workflow/controller/exec_control.go

```diff
@@ -116,7 +116,7 @@ func (woc *wfOperationCtx) killDaemonedChildren(ctx context.Context, nodeID stri
 				firstErr = err
 			}
 		}
-
+		childNode.Phase = wfv1.NodeSucceeded
```

Comment:
I don't understand this fix. Surely daemoned steps should be marked as complete during the reconciliation loop, specifically in `assessNodeStatus`?

how about make one PR for the UI fix - which I can rubber-stamp, and then follow up with the Golang changes

Reply:
> I don't understand this fix. Surely daemoned steps should be marked as complete during the reconciliation loop, specifically in assessNodeStatus?

Yes, the issue is that when the workflow ends and is marked Succeeded (such as is the case in the test in this PR), there won't be another reconciliation loop since the WorkflowStatus is now `Fulfilled`. Here, we move the responsibility of marking these nodes as successful to a reconciliation down the line, to the one that actually terminates the Pod.

> how about make one PR for the UI fix - which I can rubber-stamp, and then follow up with the Golang changes

Done https://github.com/argoproj/argo-workflows/pull/5449

Reply:
@alexec Would you like to review ☝️ before I merge?

Reply:
please merge